src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -1169,10 +1168,29 @@ class module_sram : public module_impl
     }
   }
 
-  [[nodiscard]] const std::vector<std::pair<uint64_t, uint64_t>>&
-  get_ctrlcode_addr_and_size() const override
+  void
+  fill_ert_dpu_data(uint32_t* payload) override
   {
-    return m_column_bo_address;
+    auto os_abi = m_parent.get()->get_os_abi();
+    if (os_abi == Elf_Amd_Aie2p) {
+      for (auto [addr, size] : m_column_bo_address) {
+        auto dpu = reinterpret_cast<ert_dpu_data*>(payload);
+        dpu->instruction_buffer = addr;
+        dpu->instruction_buffer_size = static_cast<uint32_t>(size);
+        dpu->chained = 0; // Always set chained as 0 for aie2p
```

Comment:
Since you are doing a loop here, is it really OK to set chained to 0? The definition of "chained" is to indicate that there are more "ert_dpu_data" followed and the loop does mean that there might be more than 1 ert_dpu_data?

Reply:
According to @larry9523 , 
**The definition of "chained" is to indicate that there are more "ert_dpu_data" followed**
only applies to aie2ps.

This is why I start this PR.